dynamic_programming/dp_bitmasking.cpp

```diff
@@ -0,0 +1,166 @@
+    std::vector<uint64_t> dp(
+        M, 0);  /// dp[i] contains max score possible by mask i
+
+    // the outer loop iterate over all possible mask till all the elements are
+    // added. the first inner loop collects index of all un-added elements.
+    // nested inner loop iterate over all the pairs formed by un added elements.
+    for (uint32_t i = 0; i < M; i++) {
+        std::vector<uint8_t> unselected_elements;
+        for (uint8_t j = 0; j < 2 * N; j++)
+            if ((i >> j) % 2 == 0)
+                unselected_elements.push_back(j);
+
+        auto total_unselected =
+            static_cast<uint8_t>(unselected_elements.size());
+        if (total_unselected % 2 == 1)
+            continue;
+
+        uint8_t total_selected = 2 * N - total_unselected;
+        uint8_t total_pairs = total_selected / 2;
+
+        for (uint8_t j1 = 0; j1 < total_unselected; j1++) {
+            for (uint8_t j2 = j1 + 1; j2 < total_unselected; j2++) {
+                uint8_t x = unselected_elements[j1],
+                        y = unselected_elements[j2];
+                uint32_t new_mask = i | (1 << x) | (1 << y);
+                dp[new_mask] =
+                    std::max(dp[new_mask], dp[i] + (total_pairs + 1) * G[x][y]);
+            }
+        }
+    }
+
+    // Max score is in the mask 2**(1<<2*N) - 1, which shows all elements are
+    // added.
+    return dp[M - 1];
+}
+
+}  // namespace dp_bitmask
+}  // namespace dynamic_programming
+
+/**
+ * @brief Self-test implementation
+ * @returns void
+ */
+static void test() {
+    // 1st test
+    const uint8_t N1 = 5;
+    std::array<uint32_t, 2 * N1> A = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+    uint64_t ans = 0;
+    ans = dynamic_programming::dp_bitmask::find_max_score(A, N1);
+    assert(ans == 55);
+
+    // 2nd test
+    const uint8_t N2 = 4;
+    std::array<uint32_t, 2 * N2> B = {1, 2, 3, 4, 5, 6, 7, 8};
+    ans = dynamic_programming::dp_bitmask::find_max_score(B, N2);
+    assert(ans == 28);
+
+    return;
+}
+
+/**
+ * @brief Function to convert argument string of digits to number
+ * @param The string of digtis that is passed in argument
+ * @return 32 bit number formed by argument
+ */
+uint32_t convert_to_num(char* s) {
+    uint32_t num = 0;
+    for (int i = 0; s[i]; i++) {
+        uint32_t digit = s[i] - '0';
+
+        // Making sure the input contains number only
+        assert(digit >= 0);
+        assert(digit <= 9);
+
+        // adding this digit at unit place.
+        num = num * 10 + digit;
+    }
+
+    // Returning finally formed number.
+    return num;
+}
+
+/**
+ * @brief Main function, runs self-test by default. User can also pass array of
+ * numbers of even length to test custom example.
+ * @param argc number of argument passed through command line
+ * @param argv is array of string arguments, user can pass array from CLI.
+ * @returns 0 on exit
+ */
+int main(int argc, char* argv[]) {
+    uint8_t N = argc - 1;

```

Comment:
You don't have to use commandline argument count and commandline array of arguments. Previous code was just fine. argv and argc is just for delaration purpose. See this for an example https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program